packages/zoe/src/contracts/loan/types.js

```diff
@@ -0,0 +1,183 @@
+ *
+ * @property {AutoswapInstance} autoswapInstance
+ *
+ * @property {PriceAuthority} priceAuthority
+ *
+ *   Used for getting the current value of collateral and setting
+ *   liquidation triggers.
+ *
+ * @property {PeriodAsyncIterable} periodAsyncIterable
+ *
+ * @property {InterestRate} interestRate
+ */
+
+/**
+ * @typedef LenderSeatProperty
+ * @property {ZCFSeat} lenderSeat
+ *
+ *   The ZCFSeat representing the lender's position in the contract.
+ */
+
+/**
+ * @typedef {LoanTerms & LenderSeatProperty} LoanConfigWithLender
+ *
+ * The loan now has a lenderSeat, which is added to the config.
+ */
+
+/**
+ * @typedef BorrowerConfigProperties
+ *
+ * @property {ZCFSeat} collateralSeat
+ *
+ *   The ZCFSeat holding the collateral in escrow after the borrower
+ *   escrows it
+ *
+ * @property {() => Amount} getDebt
+ *
+ *   A function to get the current debt
+ *
+ * @property {PromiseKit} liquidationPromiseKit
+ *
+ *   PromiseKit that includes a promise that resolves to a PriceQuote
+ *   when liquidation is triggered
+ */
+
+/**
+ * @typedef {LoanConfigWithLender & BorrowerConfigProperties } LoanConfigWithBorrower
+ *
+ * The loan has a lender, a borrower, and collateral escrowed.
+ */
+
+/**
+ * @callback ScheduleLiquidation
+ * @param {ContractFacet} zcf
+ * @param {LoanConfigWithBorrower} config
+ */
+
+/**
+ * @callback MakeLendInvitation
+ * @param {ContractFacet} zcf
+ * @param {LoanTerms} config
+ * @returns {Promise<Invitation>} lendInvitation
+ */
+
+/**
+ * @callback MakeBorrowInvitation
+ * @param {ContractFacet} zcf
+ * @param {LoanConfigWithLender} config
+ * @returns {Promise<Invitation>} borrowInvitation
+ */
+
+/**
+ * @callback MakeCloseLoanInvitation
+ * @param {ContractFacet} zcf
+ * @param {LoanConfigWithBorrower} config
+ * @returns {Promise<Invitation>} closeLoanInvitation
+ */
+
+/**
+ * Allows holder to add collateral to the contract. Exits the seat
+ * after adding.
+ *
+ * @callback MakeAddCollateralInvitation
+ * @param {ContractFacet} zcf
+ * @param {LoanConfigWithBorrower} config
+ * @returns {Promise<Invitation>} addCollateralInvitation
+ */
+
+/**
+ * @callback Liquidate
+ * @param {ContractFacet} zcf
+ * @param {LoanConfigWithBorrower} config
+ * @returns {void}
+ */
+
+/**
+ * @callback MakeDebtCalculator
+ * @param {DebtCalculatorConfig} debtCalculatorConfig
+ */
+
+/**
+ * @callback CalcInterestFn
+ * @param {number} oldDebtValue
+ * @param {number} interestRate
+ * @returns {number} interest
+ */
+
+/**
+ * @typedef {Object} DebtCalculatorConfig
+ * @property {CalcInterestFn} calcInterestFn
+ *
+ *   A function to calculate the interest, given the debt value and an
+ *   interest rate in basis points.
+ *
+ * @property {Amount} originalDebt
+ *
+ *   The debt at the start of the loan, in Loan brand
+ *
+ * @property {AmountMath} loanMath
+ *
+ *   AmountMath for the loan brand
+ *
+ * @property {PeriodAsyncIterable} periodAsyncIterable
+ *
+ *   The AsyncIterable to notify when a period has occurred
```

Comment:
I'm confused about the point of view of this description. Who is responsible for updating this iterable and who can make use of it?  Which side should this comment be aimed at?

```suggestion
 *   The AsyncIterable that will be notified when a period has occurred
```

Reply:
It is the subscription part. It will tell the contract when a period has occurred. I can try to find better language.

Reply:
Changed to ' *   Used to tell the contract when a period has occurred'